frontend/src/store/root.js

```diff
@@ -2,18 +2,15 @@ import { get } from '../utils/api'
 
 /* root application state */
 const state = {
-  loading: false,
+  error: '',
   // cached courses
-  courses: [],
-  // fire authObject
-  userAuthObject: {},
-  // our own user data
-  profile: {}
+  courses: []
 }
 
 const getters = {
-  isLoggedIn: ({ userAuthObject }) => !!userAuthObject,
-  authObject: ({ userAuthObject }) => userAuthObject
+  loading: ({ auth, course, questions, reviews, subject }) =>
+    [auth.loading, course.loading, questions.loading, reviews.loading, subject.loading].reduce((acc, curr) => acc || curr, false),
```

Comment:
Not sure what I think of this. It's quite distracting seeing the spinner at top all the time, and means we don't lean on the substates.

Reply:
i kinda put it in for testing. 
visually it's not ideal but it is actually handy for debugging - maybe i should just comment it out of the nav bar?

Reply:
could work yeah.